src/main/java/core/basesyntax/storage/impl/StorageImpl.java

```diff
@@ -0,0 +1,33 @@
+package core.basesyntax.storage.impl;
+
+import core.basesyntax.storage.Storage;
+import java.util.Collection;
+import java.util.HashMap;
+
+public class StorageImpl implements Storage {
+    private final HashMap<String, StorageEntry> entries;
+
+    public StorageImpl(HashMap<String, StorageEntry> entries) {
```

Comment:
The constructor that takes a `HashMap` parameter could lead to potential issues if the passed map is modified from outside the class. If you need to initialize the storage with pre-existing values, consider copying the data into a new map within the constructor to ensure encapsulation.